Translated post react v15 rc2 (#344)
reactjs/pt-BR.reactjs.org: content/blog/2016-03-16-react-v15-rc2.md

---

> title: "React v15.0 Release, Candidata 2"

Reviewer: Estou em dúvida como melhor traduzir isso, o que acha de?

```suggestion
title: "React v15.0 Candidato à Lançamento 2"
```

Author: achei melhor a sua sugestão, quando traduzi também fiquei na dúvida

author: [zpao]
---

> Hoje, estamos lançando um segundo candidato à versão 15. Principalmente, trata-se de 2 problemas, mas também recebemos algumas pequenas alterações de novos colaboradores, incluindo algumas melhorias em alguns de nossos novos avisos.

Reviewer: ```suggestion
Hoje, estamos lançando um segundo candidato à lançamento para a versão 15. Primariamente é para resolver 2 problemas, mas também recebemos algumas pequenas alterações de novos colaboradores, incluindo algumas melhorias em alguns de nossos novos avisos.
```

> Hoje, estamos lançando um segundo candidato à lançamento para a versão 15. Primariamente é para resolver 2 problemas, mas também recebemos algumas pequenas alterações de novos colaboradores, incluindo algumas melhorias em alguns de nossos novos avisos.

Reviewer 2: ```suggestion
Hoje, estamos lançando um segundo candidato à lançamento para a versão 15. Essencialmente é para resolver 2 problemas, mas também recebemos algumas pequenas alterações de novos colaboradores, incluindo algumas melhorias em alguns de nossos novos avisos.
```

A mudança mais urgente que foi feita é corrigir um erro em nosso novo código que remove os `<span>`s, conforme discutido na publicação original do RC1. Especificamente, temos algum código que segue um caminho diferente no IE11 e no Edge devido à velocidade de algumas operações do DOM. Houve um erro neste código que não interrompeu a otimização para os `DocumentFragment`s, resultando no texto não aparecendo. Obrigado às várias pessoas que [relataram isso](https://github.com/facebook/react/issues/6246).

> A outra alteração é no nosso código SVG. No RC1, tomamos a decisão de passar por todos os atributos diretamente. Isso levou a [alguma confusão com `class` vc `className`](https://github.com/facebook/react/issues/6211) e finalmente nos levou a reconsiderar nossa posição sobre a abordagem. Passar por todos os atributos significava que teríamos dois padrões diferentes para usar o React, onde coisas como atributos hifenizados funcionariam para SVG, mas não para HTML. No futuro, *poderemos* alterar nossa abordagem para o problema do HTML também, mas, enquanto isso é importante manter a consistência. Portanto, revertemos as alterações que atributos e expandimos a lista de propriedades SVG para incluir todos os atributos que estão na especificação. Acreditamos que temos tudo agora, mas definitivamente [nos avise](https://github.com/facebook/react/issues/1657#issuecomment-197031403) se perdemos alguma coisa. Foi e ainda é nossa intenção oferecer suporte a toda a gama de tags e atributos SVG nesta versão.

Reviewer: ```suggestion
A outra alteração é no nosso código SVG. No RC1, tomamos a decisão de passar todos os atributos diretamente. Isso levou a [alguma confusão com `class` vs `className`](https://github.com/facebook/react/issues/6211) e finalmente nos levou a reconsiderar nossa posição sobre a abordagem. Passar todos os atributos significava que teríamos dois padrões diferentes para usar o React, onde coisas como atributos hifenizados funcionariam para SVG, mas não para HTML. No futuro, *poderemos* alterar nossa abordagem para o problema do HTML também, mas, enquanto isso é importante manter a consistência. Portanto, revertemos as alterações que passavam os atributos e expandimos a lista de propriedades SVG para incluir todos os atributos que estão na especificação. Acreditamos que temos tudo agora, mas definitivamente [nos avise](https://github.com/facebook/react/issues/1657#issuecomment-197031403) se deixamos passar alguma coisa. Foi, e ainda é, nossa intenção oferecer suporte a toda a gama de tags e atributos SVG nesta versão.
```

Agradecemos novamente a todos que experimentaram o RC1 e relataram problemas. Foi extremamente importante e não poderíamos fazer isso sem a sua ajuda!

## Instalação {#installation}

Recomendamos usar o React do `npm` e usar uma ferramenta como o browserify ou o webpack para criar seu código em um único pacote. Para instalar os dois pacotes:

* `npm install --save react@15.0.0-rc.2 react-dom@15.0.0-rc.2`

> Lembre-se de que por padrão, o React executa verificações extras e fornece avisos úteis no modo de desenvolvimento. Ao implementar seu aplicativo, defina a variável de ambiente `NODE_ENV` como `production`para usar o build de produção do React, que não inclui os avisos de desenvolvimento e executa significamente mais rápido.

Reviewer: ```suggestion
Lembre-se de que por padrão, o React executa verificações extras e fornece avisos úteis no modo de desenvolvimento. Ao implantar seu aplicativo, defina a variável de ambiente `NODE_ENV` como `production`para usar o build de produção do React, que não inclui os avisos de desenvolvimento e executa significativamente mais rápido.
```

> Se você ainda não pode usar o `npm`, fornecemos versões pré-construídas do navegador para sua conveniência, que também estão disponíveis no pacode do `react` do bower.

Reviewer 2: ```suggestion
Se você ainda não pode usar o `npm`, fornecemos versões pré-construídas para navegador para sua conveniência, que também estão disponíveis no pacode do `react` do bower.
```

* **React**  

>   Dev build com avisos: <https://fb.me/react-15.0.0-rc.2.js>  

Reviewer 2: ```suggestion
  Versão para desenvolvimento com avisos: <https://fb.me/react-15.0.0-rc.2.js>  
```

>   Minified build de produção: <https://fb.me/react-15.0.0-rc.2.min.js>  

Reviewer 2: ```suggestion
  Versão minificada para produção: <https://fb.me/react-15.0.0-rc.2.min.js>  
```

* **React com Add-Ons**  

>   Dev build com avisos: <https://fb.me/react-with-addons-15.0.0-rc.2.js>  

Reviewer 2: ```suggestion
  Versão para desenvolvimento com avisos: <https://fb.me/react-with-addons-15.0.0-rc.2.js>  
```

>   Minified build de produção: <https://fb.me/react-with-addons-15.0.0-rc.2.min.js>  

Reviewer 2: ```suggestion
  Versão minificada para produção: <https://fb.me/react-with-addons-15.0.0-rc.2.min.js>  
```

* **React DOM** (inclua React na página antes do React DOM)  

>   Dev build com avisos: <https://fb.me/react-dom-15.0.0-rc.2.js>  

Reviewer 2: ```suggestion
  Versão para desenvolvimento com avisos: <https://fb.me/react-dom-15.0.0-rc.2.js>  
```

>   Minified build de produção: <https://fb.me/react-dom-15.0.0-rc.2.min.js>  

Reviewer 2: ```suggestion
  Versão minificada para produção: <https://fb.me/react-dom-15.0.0-rc.2.min.js>  
```
